tests/manage/pv_services/pvc_resize/test_resource_deletion_during_pvc_expansion.py

```diff
@@ -0,0 +1,186 @@
+
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import (
+    skipif_ocs_version, ManageTest, tier4, tier4b, ignore_leftover_label
+)
+from tests import disruption_helpers
+
+log = logging.getLogger(__name__)
+
+
+@tier4
+@tier4b
+@skipif_ocs_version('<4.5')
+@ignore_leftover_label(constants.drain_canary_pod_label)
+@pytest.mark.parametrize(
+    argnames='resource_to_delete',
+    argvalues=[
+        pytest.param(
+            'mgr', marks=pytest.mark.polarion_id('OCS-2224')
+        ),
+        pytest.param(
+            'osd', marks=pytest.mark.polarion_id('OCS-2225')
+        ),
+        pytest.param(
+            'rbdplugin', marks=pytest.mark.polarion_id('OCS-2226')
+        ),
+        pytest.param(
+            'cephfsplugin', marks=pytest.mark.polarion_id('OCS-2227')
+        ),
+        pytest.param(
+            'rbdplugin_provisioner', marks=pytest.mark.polarion_id('OCS-2228')
+        ),
+        pytest.param(
+            'cephfsplugin_provisioner',
+            marks=pytest.mark.polarion_id('OCS-2229')
+        )
+    ]
+)
+class TestResourceDeletionDuringPvcExpansion(ManageTest):
+    """
+    Tests to verify PVC expansion will be success even if rook-ceph, csi pods
+    are re-spun during the expansion
+
+    """
+    @pytest.fixture(autouse=True)
+    def setup(
+        self, multi_pvc_factory, dc_pod_factory, service_account_factory
+    ):
+        """
+        Create PVCs and pods
+
+        """
+        pvc_size = 10
+        access_modes_cephfs = [
+            constants.ACCESS_MODE_RWO, constants.ACCESS_MODE_RWX
+        ]
+        access_modes_rbd = [
+            constants.ACCESS_MODE_RWO, f'{constants.ACCESS_MODE_RWO}-Block',
+            f'{constants.ACCESS_MODE_RWX}-Block'
+        ]
+
+        pvcs_cephfs = multi_pvc_factory(
+            interface=constants.CEPHFILESYSTEM, size=pvc_size,
+            access_modes=access_modes_cephfs, status=constants.STATUS_BOUND,
+            num_of_pvc=2, timeout=90
+        )
+
+        pvcs_rbd = multi_pvc_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            project=pvcs_cephfs[0].project, size=pvc_size,
+            access_modes=access_modes_rbd,
+            status=constants.STATUS_BOUND, num_of_pvc=3, timeout=90
+        )
+        self.pvcs = pvcs_cephfs + pvcs_rbd
+
+        # Set volume mode on PVC objects
+        for pvc_obj in self.pvcs:
+            pvc_info = pvc_obj.get()
+            setattr(pvc_obj, 'volume_mode', pvc_info['spec']['volumeMode'])
+
+        sa_obj = service_account_factory(project=pvcs_cephfs[0].project)
+
+        self.pods = []
+        for pvc_obj in self.pvcs:
+            if constants.CEPHFS_INTERFACE in pvc_obj.storageclass.name:
+                interface = constants.CEPHFILESYSTEM
+            else:
+                interface = constants.CEPHBLOCKPOOL
+            # Create pods. Create 2 pods if PVC access mode is RWX
+            pod_objs = [
+                dc_pod_factory(
```

Comment:
consider using pod_factory, it can now create deploymentconfig and will also create the service account

Reply:
Done